tests/expected/shadow/unsupported_num_objects/test.rs

```diff
@@ -15,14 +15,14 @@ fn check_max_objects<const N: usize>() {
     // - the NULL pointer whose object ID is 0, and
     // - the object ID for `i`
     while i < N {
-        let x = i;
-        assert_eq!(kani::mem::pointer_object(&x as *const usize), i + 2);
+        let x: Box<usize> = Box::new(i);
```

Comment:
Why this change?

Reply:
MIR doesn't actually introduce new objects for `x` in this loop, it allocates just one object at function scope. The reason we previously considered this as producing multiple objects was through our interpretation of StorageLive as yielding a declaration. With that gone I instead had to go for an approach that would also produce fresh objects under MIR semantics.